backend/src/main/java/co/inventorsoft/scripty/service/UserServiceImpl.java

```diff
@@ -96,6 +103,38 @@ public void validateVerificationToken(final String token) {
         tokenRepository.delete(verificationToken);
     }
 
+    public void setPicture(String email, MultipartFile picture) {
+        Optional<User> user = userRepository.findByEmail(email);
+        if (!ImageTypes.contains(Files.getFileExtension(picture.getOriginalFilename()))) throw new ApplicationException("Incorrect file extension", HttpStatus.BAD_REQUEST);
+        if (user.isPresent()) {
+            Picture userPic = new Picture();
+            try {
+                userPic.setContent(picture.getBytes());
+                userPic.setExtension(picture.getContentType());
+                user.get().setPicture(userPic);
+            } catch (IOException e) {
+                throw new ApplicationException("File's empty. Please, try to use another one!", HttpStatus.BAD_REQUEST);
+            }
+        } else
+            throw new ApplicationException("User not found", HttpStatus.BAD_REQUEST);
+        userRepository.save(user.get());
+    }
+
+    public PictureDto getPicture(Long id) {
+        Optional<User> user = userRepository.findById(id);
+        PictureDto image = new PictureDto();
+
+        if (user.isPresent()) {
```

Comment:
My favorite subject - usage of `Optional`:
`userRepository.findById(id)
.flatMap(user -> Optional.ofNullable(user.getPicture()))
.map(picture -> new PictureDto(picture.getExtension(), Base64.getEncoder().encodeToString(picture.getContent())))
.orElseThrow(() -> new ApplicationException("Picture not found for user "+id));`